Security 1.0 (add jwt support)

aksoli666 opened the pull request:
add endpoint login()
created AuthenticationService, JwtAuthenticationFilter, JwtUtil

aksoli666 committed:
Security 1.0 (add jwt support)
add endpoint login()
created AuthenticationService, JwtAuthenticationFilter, JwtUtil

Elena-Bruyako commented on src/main/java/shop/controller/AuthController.java:
```diff
@@ -6,21 +6,30 @@
         return userService.register(userRegistrationRequestDto);
     }
+
+    @PostMapping("/login")
```
add Swagger annotations where are needed

Elena-Bruyako commented on src/main/java/shop/filter/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,49 @@
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (bearerToken != null && bearerToken.startsWith("Bearer ")) {
```
create constant for `"Bearer "`

Elena-Bruyako commented on src/main/java/shop/filter/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,49 @@
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (bearerToken != null && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
```
```suggestion
            return bearerToken.substring(constant.length());
```

Elena-Bruyako commented on src/main/java/shop/filter/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

Elena-Bruyako reviewed (changes requested):
Good job! Let's improve your solution!

aksoli666 committed:
Security 2.0 (add jwt support)
added swagger annotation in AuthController
created constant "Bearer " and used length constant
used HttpHeaders. Authorization instead of "Authorization"
